integration-tests/conftest.py

```diff
@@ -1,10 +1,27 @@
 import pytest
+import subprocess
 
 
+@pytest.fixture(scope="session")
+def install_katello_rpm(test_config):
+    if "satellite" in test_config.environment:
+        # install katello rpm before register system against Satellite
+        cmd = "rpm -qa 'katello*' | xargs --no-run-if-empty rpm -e"
+        ret = subprocess.check_call(cmd, shell=True)
+        
+        satellite_hostname = test_config.get("candlepin", "host")
+        cmd = "rpm -Uvh http://%s/pub/katello-ca-consumer-latest.noarch.rpm" %satellite_hostname
```

Comment:
please split this command invocation as list of arguments, ie
```python
  cmd = ["rpm", "-Uvh", ...]
```

Reply:
I do not think we should clean up katello-ca package if we test like the following 2 ways:
1.  Use a machine to test insights-client with satellite, and then re-install this machine with specified RHEL version/compose for another round of testing.
2. Use a vm to test insights-client with satellite, and then delete the vm. When start another round of testing, then create a fresh vm with specified RHEL version/compose.

When I start a testing, I prefer to use a totally new/clear system with specified RHEL version, so clean up katello-ca package is not necessary in my opinion.

Reply:
While it seems not useful on throwaway systems, it makes a difference on systems that you want to reuse again for running the tests. This is why, when using the fixtures of tools of pytest-client-tools, there are cleanup steps done after each test and at the end of the test session.

I think it is important also when re-running the test on an existing system, for example to debug failures: you want to ensure the system is "clean", so in case something changes on the server then the local system is not affected by that.

Reply:
This command is used in our code: cmd = "rpm -qa 'katello-ca-consumer*' | xargs --no-run-if-empty rpm -e"
when re-running insights-client+satellite test on an existing system, this command can cleanup the existing katello-ca(if exists) and re-install the latest one from specified satellite.

Reply:
By the way, I have run 'black' command on the file affected by this PR.

Reply:
> this command can cleanup the existing katello-ca(if exists) and re-install the latest one from specified satellite.

while this is correct, in general it's better for something (a test, in this case) to clean after itself when done

Reply:
Can we keep this idea in mind, and not to write the cleanup code for now? If we can launch regular testings when we have a jenkins job, from which we can figure out the real requirement. If we have this test requirement from real testings, we can add the cleanup code then.

Reply:
> write the cleanup code for now?

in the end, it'd be only about switching where the cleanup is done -- something like this (extremely simplified):
```python
@pytest.fixture()
def install_sat_packages()
    if is_sat:
        subprocess.check_call(["rpm", "-Uvh", ...])
    yield
    if is_sat:
        subprocess.check_call(["rpm", "-e", ...])
```
in the end, what would be the point of doing the cleanup before starting the test, if the use case you have in mind is running the tests on a newly provisioned jenkins worker? there is definitely nothing to cleanup there :)

Reply:
Ah, I see. you suggest to split my steps into 2 parts. Done as your suggested.